src/Type/Doctrine/EntityManagerFindDynamicReturnTypeExtension.php

```diff
@@ -43,12 +43,7 @@ public function getTypeFromMethodCall(
 			return $mixedType;
 		}
 
-		$type = new ObjectType($argType->getValue());
-		if ($methodReflection->getName() === 'find') {
-			$type = TypeCombinator::addNull($type);
-		}
-
-		return $type;
+		return TypeCombinator::addNull(new ObjectType($argType->getValue()));
```

Comment:
This fix may be valid for ORM, but it's invalid for MongoDB ODM: `getReference` and `getPartialReference` don't return `null` in ODM. That said, it's best to further split this extension since both methods in question aren't part of the `ObjectManager` interface.